### ISSUE TITLE
Transparent area underneath the keyboard

### ISSUE THREAD

**Issue by SWi7CHbladE** [futo-org/android-keyboard#772]
### What happened?

When navigation bar is hidden, the bottom side of the keyboard becomes transparent
![Screenshot_20241107-205420](https://github.com/user-attachments/assets/3627406c-6bdd-4957-808a-b70cf464c646)

### Keyboard Version

0.1.24-playstore

**Comment by dean-futo** [comment]
Will you please include a video? Which navigation bar are you referring to? What phone and OS are you using?

**Comment by SWi7CHbladE** [comment]
It's a motorola edge 30 on android 14

**Comment by SWi7CHbladE** [comment]
https://github.com/user-attachments/assets/f48a2202-7b43-4ce9-ab09-bb7ebde427b8

You can see that in the first half, when the navigation bar is enabled, the keyboard works fine. And when it is hidden, there is a transparent area underneath the keyboard.

**Comment by Erin0411** [comment]
i think it’s something to do with the phone’s software since it’s probably making the background of the status bar also invisible for some reason.

**Comment by UserBoi1** [comment]
I've got a Motorola Edge 40 with Android 14 and have found the same problem and also found it to really annoying.

**Comment by silkops** [comment]
Yes, same here. Moto Edge 40 Neo, Android 14. Bar under keyboard always transparent. Looks pretty ugly.

**Comment by SWi7CHbladE** [comment]
> i think it’s something to do with the phone’s software since it’s probably making the background of the status bar also invisible for some reason.

Other keyboards seem to be working well, like gboard ot typewise. It's only an issue with futo and some other keyboards.

**Comment by knitschmann** [comment]
The linked issue was closed as "won't do", whereas this one was not. May ask if there is a progress here?

Yes, this seems to be linked to **Motorola** devices (just like mine). No, this issue does not occur on Gboard, **but Swiftkey very much has this issue exactly like FUTO**. So there are other third party keyboards that face the same issue, therefor not an isolated issue.

cheers

**Comment by wertzuz** [comment]
Not solved now it's broken when its enabled

**Comment by abb128** [comment]
@wertzuz Can you explain what's broken, what's enabled, what version?

**Comment by wertzuz** [comment]
When the navigation bar on Motorola is enabled (shown) it now overlaps with the keyboard in the newest dev build with the posted fix for the opposite problem applied.

**Comment by abb128** [comment]
Would you be able to take a screenshot just so I can understand better?

**Comment by abb128** [comment]
@wertzuz I was able to reproduce it in an emulator, and made a new nightly build where it's hopefully fixed, can you let me know if that one works better?

**Comment by wertzuz** [comment]
Well now it's again transparent underneath the keyboard, when navigation at is hidden, shown works fine now. 
FUTO newest dev build: 
![Screenshot_20251005-103847.FUTO Keyboard [Dev Build].png](https://github.com/user-attachments/assets/16bfd383-5b9d-4375-9578-702484037bfa)

GBoard: 
![Screenshot_20251005-103909.FUTO Keyboard [Dev Build].png](https://github.com/user-attachments/assets/89e5b47a-5eda-4c63-b733-f71d4ef9bfbc)

**Comment by abb128** [comment]
I don't know what else I can do then, because I'm no longer specifying a transparent navbar color but it's as if Motorola is applying one anyway, and I don't have a Motorola to debug with. I'll mark it as unsolved

**Comment by wertzuz** [comment]
I think you just have to apply the first fix conditionally whether its enabled or disabled, or apply it and add extra free space before keys start
Some useful resources i found:
https://github.com/futo-org/android-keyboard/commit/1526ecb4c4f60cd60b4195817802348692563119 https://github.com/futo-org/android-keyboard/commit/4850b5d4791bee1224233d27fcf686fcdda847d1
https://github.com/Helium314/HeliBoard/issues/1818 https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1955
https://developer.android.com/reference/android/view/Window#setNavigationBarColor(int)
https://www.reddit.com/r/PocoPhones/comments/1q4jlyu/guide_to_removing_the_bottom_arrow_and_space_bar/ 
https://www.reddit.com/r/motorola/comments/1o1gia4/how_to_remove_the_bottom_bar_from_gboard_after/
